lib/sequence.js

```diff
@@ -13,15 +13,15 @@ class Sequence extends Readable {
     this.runPipelines().catch(err => this.emit('error', err))
   }
 
-  runPipelines () {
-    return this.pipelines.reduce((promise, pipeline) => {
-      return promise.then(() => {
-        const next = eventToPromise(pipeline, 'end')
-        pipeline.read()
-        return promise.then(next)
-      })
-    }, Promise.resolve())
-  };
+  async runPipelines () {
+    for (const pipeline of this.pipelines) {
+      pipeline.resume()
+
+      await promisify(finished)(pipeline)
```

Comment:
What is the `finished` import?

Reply:
It was recently added to the stream package. It's a callback version of the initial `run` function in the core package (now already replaced with `finished`). It waits for the end of a stream and also handles errors. Now, that almost everything in a pipeline is a stream, I thought it's better to use standard stream functions for consistency.